### PR TITLE
Do not use binary builds on precise

### PR DESCRIPTION
**Pull request by jimhester**
Fixes travis-ci/travis-ci#7654

### CODE REVIEW

**Comment by jimhester** [comment]
@BanzaiMan I am not sure if the condition as written is valid, in particular what is the default value of `config[:dist]` if it is not explicitly set in the `.travis.yml`?

This fix is fairly high priority to deploy, as all R builds on precise with `sudo: true` will fail when they try to retrieve the binary packages with the current code.

**Comment by jimhester** [comment]
I don't believe the failing integration tests are related to my changes, but let me know if this is not the case.

File: lib/travis/build/script/r.rb [hunk 1]
```diff
@@ -324,7 +324,7 @@ def r_binary_install(packages)
           return if packages.empty?
           packages = Array(packages)
           if config[:os] == 'linux'
-            unless config[:sudo]
+            unless config[:sudo] and config[:dist] != 'precise'
```
Comment (BanzaiMan) [inline lib/travis/build/script/r.rb]: Eschew compound conditions with `unless`. It is easier to understand:

```ruby
if !config[:sudo] or config[:dist] == 'precise'
```
than
```ruby
unless config[:sudo] and config[:dist] != 'precise'
```

Also, this new criterion should be mentioned in the message below.

Comment (jimhester) [inline lib/travis/build/script/r.rb]: Thanks for the review, just pushed with the requested changes!

**Pull request merged**